keps/sig-node/4009-add-cdi-devices-to-device-plugin-api/README.md

```diff
@@ -0,0 +1,867 @@
+documentation such as release notes or a development roadmap. It should be
+possible to collect this information before implementation begins, in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+This KEP proposes extending the Device Plugin API, adding a field to specify
+Container Device Interface (CDI) devices IDs in the `AllocateResponse`. This
+supplements the existing fields such as annotations and allows device plugin
+implementations to uniquely specify devices using their fully-qualified CDI
+devices names.
+
+The recent addition of CDI device IDs to the CRI structures in [KEP-3731](https://github.com/kubernetes/enhancements/pull/3731)
+
+allow these IDs to be forwarded to the CRI runtimes in a secure manner. Although
+these changes were motivated by [KEP-3063](https://github.com/kubernetes/enhancements/issues/3063), adding support for these fields to the
+existing device plugin API allows this mechanism to also be used for devices
+supported by these plugins.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+The Container Device Inteface (CDI) provides a standard mechanism for device
+vendors to describe what is required to provide access to a specific resource
+such as a GPU. These resources can be uniquely identified using a
+fully-qualified device name.
+
+#3731 extends the CRI to provide a well-define mechanism for forwarding such
+requests to CRI runtimes such as Containerd and Cri-o, which have already
+been extended to accept CDI device requests (currently as container annotations)
+, and to use the associated CDI specifications to ensure that the required
+modifications are made to the OCI runtime specification for a container being
+launched.
+
+The addition of the field to the device plugin API allows this same logic to be
+used to make devices available to containers without requiring that users move
+to a Dynamic Resource Allocation (DRA) based implementation.
```

Comment:
If this is really the case, it would probably makes sense to explain why CRI field is a requirement and annotations can't be used for the same purpose.

Reply:
I have responded to your comment https://github.com/kubernetes/enhancements/pull/4011/files#r1221031326